src/applications/caregivers/utils/helpers.js

```diff
@@ -1,150 +1,112 @@
-        /* if user submits a document via upload page, add the guid to the formData
-          otherwise delete object and move on to next keys */
-        if (isEmpty(data[key]) || data.signAsRepresentativeYesNo !== 'yes') {
-          return delete sortedDataByChapter.poaAttachmentId;
-        }
-
-        const documentUpload = data[key][0].guid;
-        sortedDataByChapter.poaAttachmentId = documentUpload;
-      } else if (key === 'signAsRepresentativeYesNo') {
-        if (data[key] === 'yes') {
-          sortedDataByChapter.signAsRepresentative = true; // sign as representative
-        } else {
-          sortedDataByChapter.signAsRepresentative = false; // sign as veteran
-        }
-      } else if (key.includes(dataPrefix)) {
-        // if preferredFacility grab the nested "plannedClinic" value, and surface it
-        if (key === 'veteranPreferredFacility') {
-          sortedDataByChapter[chapterName] = {
-            ...sortedDataByChapter[chapterName],
-            plannedClinic: data[key].plannedClinic,
-          };
-        } else {
-          // otherwise just remove the prefix, and populate chapter object
-          const keyWithoutPrefix = lowerCaseFirstLetter(
-            key.split(dataPrefix)[1],
+    const dataToReturn = dataKeys.reduce((acc, key) => {
+      const keyWithoutPrefix = camelCase(key.split(dataPrefix)[1]);
+
+      // map the home address to the mailing address, if applicable
+      if (key.includes('HomeSameAsMailingAddress')) {
+        if (formData[key]) {
+          acc.mailingAddress = omit(
+            formData[`${dataPrefix}Address`],
+            'country',
           );
-          sortedDataByChapter[chapterName] = {
-            ...sortedDataByChapter[chapterName],
-            [keyWithoutPrefix]: data[key],
-          };
         }
+        return acc;
+      }
+
+      // omit country from address fields, if applicable
+      if (key.endsWith('Address')) {
+        acc[keyWithoutPrefix] = omit(formData[key], 'country');
+        return acc;
       }
 
-      // returning null due to "array-callback-return" eslint rule
-      return null;
-    });
+      // otherwise just populate form data
+      acc[keyWithoutPrefix] = formData[key];
+      return acc;
+    }, {});
 
-    return sortedDataByChapter;
+    return {
+      [chapterName]: dataToReturn,
+    };
+  };
+
+  // map the form data related to signing as representative
+  const buildRespresentativeData = () => {
+    const {
+      signAsRepresentativeYesNo,
+      signAsRepresentativeDocumentUpload,
+    } = formData;
+    const signAsRepresentative = signAsRepresentativeYesNo === 'yes';
+    const dataToReturn = { signAsRepresentative };
+
+    if (!isEmpty(signAsRepresentativeDocumentUpload) && signAsRepresentative) {
+      dataToReturn.poaAttachmentId = signAsRepresentativeDocumentUpload[0].guid;
+    }
+
+    return dataToReturn;
   };
 
   const remappedData = {
     ...form,
     data: {
```

Comment:
There was no reason to pass the form data to the method, since it was already globally available to the method.